Sources/SwiftSyntaxBuilderGeneration/Templates/BuildableCollectionNodesFile.swift

```diff
@@ -0,0 +1,248 @@
+              secondName: .identifier("lists"),
+              colon: .colon,
+              type: ArrayType(elementType: type.expressibleAs)
+            )
+          )
+        )
+      ) {
+        SequenceExpr {
+          "elements"
+          AssignmentExpr()
+          FunctionCallExpr(MemberAccessExpr(base: "lists", name: "flatMap"), trailingClosure: ClosureExpr {
+            MemberAccessExpr(
+              base: FunctionCallExpr(MemberAccessExpr(base: "$0", name: "create\(type.buildable)")),
+              name: "elements"
+            )
+          })
+        }
+      }
+
+      InitializerDecl(
+        modifiers: [TokenSyntax.public],
+        signature: FunctionSignature(
+          input: ParameterClause(
+            parameterList: FunctionParameter(
+              firstName: .identifier("arrayLiteral").withTrailingTrivia(.space),
+              secondName: .identifier("elements"),
+              colon: .colon,
+              type: elementType.expressibleAs,
+              ellipsis: .ellipsis
+            )
+          )
+        )
+      ) {
+        FunctionCallExpr(MemberAccessExpr(base: "self", name: "init")) {
+          TupleExprElement(expression: "elements")
+        }
+      }
+
+      FunctionDecl(
+        modifiers: [TokenSyntax.public],
+        identifier: .identifier("build\(type.baseName)"),
+        signature: FunctionSignature(
+          input: createFormatLeadingTriviaParameters(withDefaultTrivia: true),
+          output: type.syntax
+        )
+      ) {
+        VariableDecl(
+          .let,
+          name: "result",
+          initializer: FunctionCallExpr(MemberAccessExpr(base: "SyntaxFactory", name: "make\(type.baseName)")) {
+            if elementType.isToken {
+              TupleExprElement(expression: "elements")
+            } else {
+              TupleExprElement(
+                expression: FunctionCallExpr(MemberAccessExpr(base: "elements", name: "map"), trailingClosure: ClosureExpr {
+                  FunctionCallExpr(MemberAccessExpr(base: "$0", name: "build\(elementType.baseName)")) {
+                    TupleExprElement(label: "format", expression: "format")
+                    TupleExprElement(
+                      label: "leadingTrivia",
+                      expression: node.elementsSeparatedByNewline
+                        ? SequenceExpr {
+                            MemberAccessExpr(base: "Trivia", name: "newline")
+                            BinaryOperatorExpr("+")
+                            FunctionCallExpr(MemberAccessExpr(base: "format", name: "_makeIndent"))
+                          }
+                        : "nil"
+                    )
+                  }
+                })
+              )
+            }
+          }
+        )
+        IfStmt(
+          conditions: OptionalBindingCondition(
+            letOrVarKeyword: .let,
+            pattern: "leadingTrivia",
+            initializer: "leadingTrivia"
+          )
+        ) {
```

Comment:
Something I thought about while working on this was whether we might want to remove this `if`-statement and replace e.g.

```swift
public func buildTupleExprElementList(format: Format, leadingTrivia: Trivia? = nil) -> TupleExprElementListSyntax {
  ...
  if let leadingTrivia = leadingTrivia {
    return result.withLeadingTrivia(leadingTrivia + (result.leadingTrivia ?? []))
  } else {
    return result
  }
}
```

with

```swift
public func buildTupleExprElementList(format: Format, leadingTrivia: Trivia? = nil) -> TupleExprElementListSyntax {
  ...
  let combinedLeadingTrivia = (leadingTrivia ?? []) + (result.leadingTrivia ?? [])
  return result.withLeadingTrivia(combinedLeadingTrivia)
}
```

Although we would deviate from the original translation there, the latter solution feels more consistent with `BuildableNodes` (which takes a similar approach) and would present a potential opportunity to unify these implementations by factoring the syntax generation into a utility function.

Reply:
That sounds like a good idea to me 👍